csharp/src/Ice/Properties.cs

```diff
@@ -164,17 +454,371 @@ public interface Properties
     ///  <param name="options">The command-line options.
     ///  </param>
     /// <returns>The command-line options that do not start with one of the listed prefixes, in their original order.</returns>
-    string[] parseIceCommandLineOptions(string[] options);
+    public string[] parseIceCommandLineOptions(string[] options)
+    {
+        string[] args = options;
+        foreach (string? name in PropertyNames.clPropNames)
+        {
+            args = parseCommandLineOptions(name, args);
+        }
+        return args;
+    }
 
     /// <summary>
     /// Load properties from a file.
     /// </summary>
     /// <param name="file">The property file.</param>
-    void load(string file);
+    public void load(string file)
+    {
+        try
+        {
+            using var sr = new StreamReader(file);
+            parse(sr);
+        }
+        catch (IOException ex)
+        {
+            var fe = new FileException(ex);
+            fe.path = file;
+            throw fe;
+        }
+    }
 
     /// <summary>
     /// Create a copy of this property set.
     /// </summary>
     /// <returns>A copy of this property set.</returns>
-    Properties ice_clone_();
+    public Properties ice_clone_()
+    {
+        var clonedProperties = new Properties();
+        lock (this)
+        {
+            foreach (KeyValuePair<string, PropertyValue> entry in _properties)
+            {
+                clonedProperties._properties[entry.Key] = entry.Value.Clone();
+            }
```

Comment:
I think it looks a bit nicer using a tuple
```suggestion
            foreach ((string key, PropertyValue value) in _properties)
            {
                clonedProperties._properties[key] = value.Clone();
            }
```